pkg/manager/member/pd_failover.go

```diff
@@ -125,6 +125,12 @@ func (pf *pdFailover) tryToMarkAPeerAsFailure(tc *v1alpha1.TidbCluster) error {
 		if pdMember.LastTransitionTime.IsZero() {
 			continue
 		}
+		if !pf.isPodDesired(tc, podName) {
```

Comment:
Should we also delete the non desired members in the failure members as done in https://github.com/pingcap/tidb-operator/pull/2560/files#diff-cff8f7143431f3d6302182e300a81909R58?

Reply:
I'm not sure if it's necessary right now since all failure members will be cleared. anyway, we can revise it later.